packages/systemd/9004-core-add-separate-timeout-for-system-shutdown.patch

```diff
@@ -0,0 +1,66 @@
+From 4d11f5d502ca4a61c491681cdfd99ebe24e3f58c Mon Sep 17 00:00:00 2001
+From: Ben Cressey <bcressey@amazon.com>
+Date: Tue, 10 Mar 2020 20:30:10 +0000
+Subject: [PATCH 9004/9004] core: add separate timeout for system shutdown
+
+There is an existing setting for this (DefaultTimeoutStopUSec), but
+changing it has no effect because `reset_arguments()` is called just
+before `become_shutdown()`, which sets everything back to the stock
+values.
+
+It's also awkward to use a single timeout value for stopping normal
+services, and for cleaning up stray processes after most of userspace
+has gone away.
+
+Signed-off-by: Ben Cressey <bcressey@amazon.com>
+---
+ src/basic/def.h | 3 +++
+ src/core/main.c | 4 +++-
+ 2 files changed, 6 insertions(+), 1 deletion(-)
+
+diff --git a/src/basic/def.h b/src/basic/def.h
+index 970654a..b02f6f0 100644
+--- a/src/basic/def.h
++++ b/src/basic/def.h
+@@ -13,6 +13,9 @@
+  * the watchdog pings will keep the loop busy. */
+ #define DEFAULT_EXIT_USEC (30*USEC_PER_SEC)
+ 
++/* The default time after which processes are killed on shutdown. */
++#define DEFAULT_TIMEOUT_SHUTDOWN_USEC (5*USEC_PER_SEC)
```

Comment:
Do we want this as short as 5 seconds?  The default timeout for `docker stop` on the command line is 10s and the default in the API is 30s (which is also the default used by ECS).

Reply:
In most scenarios the operator should be draining workloads off the node before we get to this point. That's always going to be better since more specific policy and knowledge about the workload can be applied.

The difficulty here is that we lack context. We don't know if a process is doing anything useful in response to `SIGTERM`: it could be ignoring it (in which case shorter timeout is better), it could be doing a lot of work (in which case no timeout would be best), or it could be wedged (somewhere in between).

I like the shorter value. For me the appeal is that it takes a node that's almost certainly not doing useful work and either brings it back that much faster or allows a replacement to be launched.

I'm OK with increasing the timeout if we hear that the lower values are a problem.

Reply:
I can buy that.  I would bias toward 10s to try and give at least a little consideration to the default `docker stop` command-line operation, but I agree that we can increase this later if we need to.

Reply:
Makes sense. I'll bump the various timeouts to 10 seconds.